flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/hint/FlinkHintStrategies.java

```diff
@@ -62,8 +62,10 @@ public static HintStrategyTable createHintStrategyTable() {
                 // internal join hint used for alias
                 .hintStrategy(
                         FlinkHints.HINT_ALIAS,
-                        // currently, only join hints care about query block alias
-                        HintStrategy.builder(HintPredicates.JOIN)
+                        // currently, only correlate&join hints care about query block alias
+                        HintStrategy.builder(
+                                        HintPredicates.or(
+                                                HintPredicates.CORRELATE, HintPredicates.JOIN))
                                 .optionChecker(fixedSizeListOptionChecker(1))
```

Comment:
the fixedSizeListOptionChecker is not applicable to CORRELATE  (actually the lookup join which only accepts kvOptions)

Reply:
the fixedSizeListOptionChecker is used to validate alias hint, not lookup hint.

Reply:
sorry, I'm not quite understand what's wrong with this piece of code, could u give me more suggestions about it?

Reply:
sorry, my misunderstood here, this optionChecker is ok.